GG/src/Font.cpp

```diff
@@ -527,9 +722,216 @@ Font::FormattingTag::FormattingTag(bool close) :
+        m_text_elements.push_back(element);
+    }
+
+    /** Add a text element.  Any whitespace in this text element will be non-breaking.*/
+    void AddText(const std::string& text)
+    {
+        m_are_widths_calculated = false;
+
+        boost::shared_ptr<Font::TextElement> element(new Font::TextElement(false, false));
+        size_t begin = m_text.size();
+        size_t end = m_text.append(text).size();
+        element->text = Substring(m_text,
+                                  boost::next(m_text.begin(), begin),
+                                  boost::next(m_text.begin(), end));
+        m_text_elements.push_back(element);
+    }
+
+    /** Add a white space element.*/
+    void AddWhitespace(const std::string& whitespace)
+    {
+        m_are_widths_calculated = false;
+
+        boost::shared_ptr<Font::TextElement> element(new Font::TextElement(true, false));
+        size_t begin = m_text.size();
+        size_t end = m_text.append(whitespace).size();
+        element->text = Substring(m_text,
+                                  boost::next(m_text.begin(), begin),
+                                  boost::next(m_text.begin(), end));
+        m_text_elements.push_back(element);
+    }
+
+    /** Add a white space element.*/
+    void AddNewline()
+    {
+        m_are_widths_calculated = false;
+
+        boost::shared_ptr<Font::TextElement> element(new Font::TextElement(false, true));
+        m_text_elements.push_back(element);
+    }
+
+    /** Add open color tag.*/
+    void AddOpenTag(const Clr& color) {
+        std::vector<std::string> params;
+        params.push_back(boost::lexical_cast<std::string>(static_cast<int>(color.r)));
+        params.push_back(boost::lexical_cast<std::string>(static_cast<int>(color.g)));
+        params.push_back(boost::lexical_cast<std::string>(static_cast<int>(color.b)));
+        params.push_back(boost::lexical_cast<std::string>(static_cast<int>(color.a)));
+
+        AddOpenTag("rgba", &params);
+    }
+
+private:
+    const Font& m_font;
+    std::string m_text;
+    std::vector<boost::shared_ptr<TextElement> > m_text_elements;
+    bool m_are_widths_calculated;
+};
+
+
+Font::TextAndElementsAssembler::TextAndElementsAssembler(const Font& font) :
+    pimpl(new TextAndElementsAssemblerImpl(font))
+{}
+// Required because Impl is defined here
+Font::TextAndElementsAssembler::~TextAndElementsAssembler()
+{}
+const std::string& Font::TextAndElementsAssembler::Text()
+{ return pimpl->Text(); }
+
+const std::vector<boost::shared_ptr<Font::TextElement> >& Font::TextAndElementsAssembler::Elements()
+{ return pimpl->Elements(); }
+
+Font::TextAndElementsAssembler& Font::TextAndElementsAssembler::AddOpenTag(const std::string& tag) {
+    pimpl->AddOpenTag(tag);
+    return *this;
+}
+
+Font::TextAndElementsAssembler& Font::TextAndElementsAssembler::AddOpenTag(
+    const std::string& tag, const std::vector<std::string>& params) {
```

Comment:
especially after multi-line function name line, put { on newline. also do so here and around here to be consistent